src/main/scala/org/opennetworkinsight/proxy/ProxyPostLDA.scala

```diff
@@ -54,10 +55,29 @@ object ProxyPostLDA {
     val scoredDF : DataFrame = score(sc, rawDataDF, topicCount, ipToTopicMix, wordsToProbPerTopic)
 
     val filteredDF = scoredDF.filter("score < " + threshold)
-    logger.info("Persisting data")
 
-    val scored = filteredDF.sort("score").rdd.map(_.mkString(",")).saveAsTextFile(resultsFilePath)
+    val count = filteredDF.count
+
+    val takeCount  = if (topK == -1 || count < topK) {
+      count.toInt
+    } else {
+      topK
+    }
+
+    val scoreIndex = filteredDF.schema.fieldNames.indexOf("score")
+
+    class RowOrderByLeastScoreToTop() extends Ordering[Row] {
+      def compare(row1: Row, row2: Row) = row2.getDouble(scoreIndex).compare(row1.getDouble(scoreIndex))
+    }
+
+    implicit val rowOrdering = new RowOrderByLeastScoreToTop()
+    val topRows : Array[Row] = filteredDF.rdd.top(takeCount)
+
+    val outRDD = sc.parallelize(topRows).sortBy(row => row.getDouble(scoreIndex))
```

Comment:
Not big deal, just that variable names are the same as DNS and Flow but names vary from the first two to Proxy. 
class RowOrderByLeast... vs class OderByLeast...
rowOrdering vs ordering
outRDD vs outputRDD